### PR TITLE
Bump native-platform: use new macOS watcher, take 2

### PR DESCRIPTION
**Pull request by lptr**
Re-try https://github.com/gradle/gradle/pull/13785

PRs merged since `0.22-milestone-4`:

- https://github.com/gradle/native-platform/pull/238
- https://github.com/gradle/native-platform/pull/241
- https://github.com/gradle/native-platform/pull/243

### CODE REVIEW

File: subprojects/language-scala/src/integTest/groovy/org/gradle/language/scala/CachedPlatformScalaCompileIntegrationTest.groovy [hunk 1]
```diff
@@ -25,6 +25,10 @@ class CachedPlatformScalaCompileIntegrationTest extends AbstractCachedCompileInt
     String compilationTask = ':compileMainJarMainScala'
     String compiledFile = "build/classes/main/jar/Person.class"
 
+    def setup() {
+        executer.withFullDeprecationStackTraceDisabled()
```
Comment (wolfs) [inline subprojects/language-scala/src/integTest/groovy/org/gradle/language/scala/CachedPlatformScalaCompileIntegrationTest.groovy:29]: Why is this necessary?

File: subprojects/distributions-dependencies/distributions-dependencies.gradle.kts [hunk 2]
```diff
@@ -118,7 +118,7 @@ dependencies {
         api(libs.maven3WagonHttpShared) { version { strictly(mavenWagonVersion) }}
         api(libs.maven3WagonProviderApi) { version { strictly(mavenWagonVersion) }}
         api(libs.minlog)                { version { strictly("1.2") }}
-        api(libs.nativePlatform)        { version { strictly("0.22-milestone-4") }}
+        api(libs.nativePlatform)        { version { strictly("0.22-snapshot-20200715094808+0000") }}
```
Comment (wolfs) [inline subprojects/distributions-dependencies/distributions-dependencies.gradle.kts:121]: Can you publish a snapshot from native platform's `master` branch?

File: subprojects/language-scala/src/integTest/groovy/org/gradle/language/scala/CachedPlatformScalaCompileIntegrationTest.groovy (hunk 1 above)
Comment (lptr) [reply subprojects/language-scala/src/integTest/groovy/org/gradle/language/scala/CachedPlatformScalaCompileIntegrationTest.groovy:29]: It's not necessary, but it's close to undebuggable when you have dozens of stacktraces to wade through.

File: subprojects/distributions-dependencies/distributions-dependencies.gradle.kts (hunk 2 above)
Comment (lptr) [reply subprojects/distributions-dependencies/distributions-dependencies.gradle.kts:121]: No, from my branch. Is this important?

Comment (wolfs) [reply subprojects/distributions-dependencies/distributions-dependencies.gradle.kts:121]: I find it easier if the snapshots we use are from native platform's `master` branch, so it is easier to check what the diff between two snapshots is. I think the rule should be that we use snapshots from `master` on Gradle's `master` branch.
